crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -245,6 +246,11 @@ impl BaseClient {
             )
             .await?;
 
+        // This caches the latest event when it comes through via sync.
+        // We probably also want to do this later, if it was encrypted, and now it has
+        // been decrypted.
+        room_info.latest_event = choose_event_to_cache(&timeline.events);
```

Comment:
We'll need to do this for non-SS responses as well.

Reply:
If we feature-gate `latest_event` behind `feature = "experimental-sliding-sync"`, it's fine for now to have it only for Sliding Sync. Thoughts?

Reply:
We went for the feature gate.